dawn/src/dawn/CodeGen/Cuda-ico/CudaIcoCodeGen.cpp

```diff
@@ -982,83 +968,99 @@ void CudaIcoCodeGen::generateAllAPIVerifyFunctions(
                                    /*indent level*/ 0, onlyDecl);
 
     // TODO to be removed
-    verifyAPI.addArg("const int edge_start_idx_c");
-    verifyAPI.addArg("const int edge_end_idx_c");
+    verifyAPI.addArg("const int start_idx");
+    verifyAPI.addArg("const int end_idx");
     // ENDTODO
-    verifyAPI.addArg("const int " + getDenseSizeName(dawn::ast::LocationType::Edges));
-    verifyAPI.addArg("const int " + getDenseSizeName(dawn::ast::LocationType::Cells));
-    verifyAPI.addArg("const int " + getDenseSizeName(dawn::ast::LocationType::Vertices));
     verifyAPI.addArg("const int k_size");
 
-    // TODO implement case in which IntendKind is InputOutput
     for(auto fieldID : getUsedFields(stencil, {dawn::iir::Field::IntendKind::Output})) {
       verifyAPI.addArg("const ::dawn::float_type *" +
                        stencilInstantiation->getMetaData().getNameFromAccessID(fieldID) + "_dsl");
       verifyAPI.addArg("const ::dawn::float_type *" +
                        stencilInstantiation->getMetaData().getNameFromAccessID(fieldID));
     }
+    for(auto fieldID : getUsedFields(stencil, {dawn::iir::Field::IntendKind::InputOutput})) {
+      verifyAPI.addArg("const ::dawn::float_type *" +
+                       stencilInstantiation->getMetaData().getNameFromAccessID(fieldID));
+      verifyAPI.addArg("const ::dawn::float_type *" +
+                       stencilInstantiation->getMetaData().getNameFromAccessID(fieldID) +
+                       "_before");
+    }
     verifyAPI.addArg("const int iteration");
     verifyAPI.finishArgs();
 
     // TODO to be removed
     runAndVerifyAPI.addArg("dawn::GlobalGpuTriMesh *mesh");
     runAndVerifyAPI.addArg("const int k_size");
-    runAndVerifyAPI.addArg("const int edge_start_idx_c");
-    runAndVerifyAPI.addArg("const int edge_end_idx_c");
+    runAndVerifyAPI.addArg("const int start_idx");
+    runAndVerifyAPI.addArg("const int end_idx");
     // ENDTODO
     addGlobalsArgs(globalsMap, runAndVerifyAPI);
     for(auto accessID : stencilInstantiation->getMetaData().getAPIFields()) {
       runAndVerifyAPI.addArg("::dawn::float_type *" +
                              stencilInstantiation->getMetaData().getNameFromAccessID(accessID));
     }
+    for(auto fieldID : getUsedFields(stencil, {dawn::iir::Field::IntendKind::InputOutput})) {
+      runAndVerifyAPI.addArg("const ::dawn::float_type *" +
+                             stencilInstantiation->getMetaData().getNameFromAccessID(fieldID) +
+                             "_before");
+    }
     runAndVerifyAPI.finishArgs();
 
     if(!onlyDecl) {
       const auto& fieldInfos = stencil.getOrderedFields();
 
       verifyAPI.startBody();
       verifyAPI.addStatement("using namespace std::chrono");
+      verifyAPI.addStatement("auto &mesh = " + fullStencilName + "::" + "mesh_");
```

Comment:
```suggestion
      verifyAPI.addStatement("const auto &mesh = " + fullStencilName + "::" + "getMesh()");
```
`getMesh()` would be the const getter to be added